stores/amazon.py

```diff
@@ -499,20 +396,27 @@ def finalize_order_button(self, test, retry=0):
                 log.info(
                     "Couldn't find button after 3 retries. Open a GH issue for this."
                 )
+                self.take_screenshot("finalize-order-button-fail")
         return returnVal
 
     def wait_for_order_completed(self, test):
         if not test:
-            self.check_if_captcha(self.wait_for_pages, ORDER_COMPLETE_TITLES)
+            try:
+                self.check_if_captcha(self.wait_for_pages, ORDER_COMPLETE_TITLES)
+            except:
+                log.error("error during order completion")
```

Comment:
```self.save_screenshot("order-failed")```

this way we have an idea of why it failed since this should be the absolute final page we see

Reply:
yeah this whole thing needs to be cleaned up and be made consistent. I'll put one there.